README.md

```diff
@@ -1 +1,292 @@
-# node-modules-patterns
\ No newline at end of file
+# Module Patterns in Node
+
+## Overview
+
+Imagine a scenario where you have an application with 2,000 lines of code. All the code is in one file. That's a nightmare to navigate around and also difficult for different developers to collaborate without merge conflicts! What if you split the logic into 20 files which each file having 100 lines of code roughly. Much better right? So what's the best way to do that? The answer is modules!
+
+Modules allow developers to build better systems because the code can be separated into different files which are grouped into packages or modules. You can write your own module or use a module from an open-source community.
+
+This lesson will cover the most common patterns used to create and export a Node module.
+
+## Objectives
+
+1. Define different patterns for Node modules
+1. Describe module.exports= {}
+1. Describe module.exports=function(){}
```

Comment:
I think it's really confusing to list these separately, as they're essentially variations on a theme (`function`'s has `Object` in its prototype chain, after all).

Instead, I think the takeaway from objectives is that you can export an entire `Object` with `module.exports` or you can export individual properties with `exports.prop = {}`.